superset/commands/database/sync_permissions.py

```diff
@@ -115,6 +116,11 @@ def validate(self) -> None:
             try:
                 alive = ping(engine)
             except Exception as err:
+                if (
+                    self.db_connection.is_oauth2_enabled()
+                    and self.db_connection.db_engine_spec.needs_oauth2(err)
+                ):
```

Comment:
In the near future we should DRY this code, we use it a lot.

Reply:
That's a good callout. I'll merge this one for now but I'll keep this in mind.